File: engine/model/src/main/scala/tools/packet/character/UpdateSkill.scala
package tools.packet.character

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class UpdateSkill(private var _skillId: Int, private var _level: Int, private var _masterLevel: Int,
                  private var _expiration: Long) extends PacketInput {
  def skillId: Int = _skillId

  def level: Int = _level

  def masterLevel: Int = _masterLevel

  def expiration: Long = _expiration

  override def opcode(): SendOpcode = SendOpcode.UPDATE_SKILLS
}

File: engine/engine-model/src/main/java/tools/packet/character/UpdateSkill.java
package tools.packet.character;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record UpdateSkill(int skillId, int level, int masterLevel, long expiration) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.UPDATE_SKILLS;
   }
}